src/test/bufferlist.cc

```diff
@@ -1848,6 +1848,22 @@ TEST(BufferList, rebuild_aligned_size_and_memory) {
   EXPECT_TRUE(bl.is_aligned(SIMD_ALIGN));
   EXPECT_TRUE(bl.is_n_align_sized(BUFFER_SIZE));
   EXPECT_EQ(3U, bl.get_num_buffers());
+
+  {
+    /* bug replicator, to test rebuild_aligned_size_and_memory() in this
+     * special scenarios */
```

Comment:
Nit: ... in the scenario where the first bptr is both size and memory aligned and the second is 0-length.